### PR TITLE
Create `SuggestBox` component

### PR DESCRIPTION
**Pull request by koesie10**
This creates a `SuggestBox` component that can be used for a VS Code-like autocomplete on a list of options.

https://github.com/github/vscode-codeql/assets/1112623/70cc0586-d734-4110-906a-6a36bc97f4df

This does not yet wire it up to anything, nor does it use the functions introduced in #3218. This component is not specific to access paths and could be used for any list of options with follow-up options.

The component uses [`@floating-ui/react`](https://floating-ui.com/docs/react) to correctly position the suggestions and to implement accessibility correctly. Documentation about this can be found on [the Floating UI docs](https://floating-ui.com/docs/react). The example most like this implementation is the [Autocomplete (combobox)](https://codesandbox.io/s/fragrant-water-bsuirj?file=/src/App.tsx) example.

To test this, you can use Storybook. There are also unit tests for most of the functionality of the component.

## Checklist

- [ ] [CHANGELOG.md](https://github.com/github/vscode-codeql/blob/main/extensions/ql-vscode/CHANGELOG.md) has been updated to incorporate all user visible changes made by this pull request.
- [ ] Issues have been created for any UI or other user-facing changes made by this pull request.
- [ ] _[Maintainers only]_ If this pull request makes user-facing changes that require documentation changes, open a corresponding docs pull request in the [github/codeql](https://github.com/github/codeql/tree/main/docs/codeql/codeql-for-visual-studio-code) repo and add the `ready-for-doc-review` label there.

### CODE REVIEW

File: extensions/ql-vscode/src/view/common/SuggestBox/SuggestBox.tsx [hunk 1]
```diff
@@ -0,0 +1,236 @@
...
+import { VSCodeTextField } from "@vscode/webview-ui-toolkit/react";
+import type { Option } from "./options";
+import { findMatchingOptions } from "./options";
+import { SuggestBoxItem } from "./SuggestBoxItem";
+
+const Input = styled(VSCodeTextField)`
+  width: 430px;
+
+  font-family: var(--vscode-editor-font-family);
+`;
```
Comment (shati-patel) [inline extensions/ql-vscode/src/view/common/SuggestBox/SuggestBox.tsx:26]: Should this have a specified background colour too? When I tested in Storybook, the light themes looked like this: 
image

**Review by shati-patel — COMMENTED** [review]
I mainly played around with this in Storybook, since the code and tests look pretty comprehensive. One question, but otherwise LGTM!

File: extensions/ql-vscode/src/view/common/SuggestBox/SuggestBox.tsx (hunk 1 above)
Comment (koesie10) [reply extensions/ql-vscode/src/view/common/SuggestBox/SuggestBox.tsx:26]: That seems to be a bug in the Storybook theme, it does work correctly when rendered in VS Code. You can see the same behavior in for example the [Variant analysis story](http://localhost:6006/?path=/story/variant-analysis-variant-analysis--full-example&globals=vscodeTheme:light).

Comment (shati-patel) [reply extensions/ql-vscode/src/view/common/SuggestBox/SuggestBox.tsx:26]: Ah, I see! That's annoying, but good to know that it's fine in VS Code 👌🏽

**Review by shati-patel — APPROVED** [review]
🚢

**Pull request merged**